Controllers/OrdersController.cs

```diff
@@ -0,0 +1,109 @@
+              {
+                  Id = o.Id,
+                  OrderTime = o.OrderTime,
+                  DeliveryTime = o.DeliveryTime,
+                  AddressFrom = o.AddressFrom,
+                  AddressTo = o.AddressTo,
+                  Status = o.Status,
+                  Employee = Context.Employees.Where(y => y.PersonalCode == o.Employee.PersonalCode).First().FirstName
+              }).ToListAsync();
+            return Ok(new GetOrdersDTO<ProductBalance>(orders));
+        }
+
+        [HttpGet("ordersf")] //return a full list of orders and products
+        public async Task<ActionResult<IEnumerable<GetOrdersDTO<ProductBalanceInterDTO>>>> GetOrders2()
+        {
+            if (!IsValidApiRequest())
+            {
+                return ApiBadRequest("Invalid Headers!");
+            }
+
+            var orders = await Context.Order
+                .Select(o => new OrdersDTO<ProductBalanceInterDTO>(){
+                    Id = o.Id,
+                    OrderTime = o.OrderTime,
+                    DeliveryTime = o.DeliveryTime,
+                    AddressFrom = o.AddressFrom,
+                    AddressTo = o.AddressTo,
+                    Status = o.Status,
+                    Products = Context.ProductBalance
+                                .Where(x => x.Order.Id == o.Id)
+                                .Select(x => new ProductBalanceInterDTO() { 
+                                        Id = x.Id,
+                                        ExpirationDate = x.ExpirationDate,
+                                        Price = x.Price,
+                                        Medicament = Context.Medicaments.Where(y => y.Id == x.Id).First().Name,
+                                        Transaction = Context.Transaction.Where(y => y.Id == x.Id)
+                                                            .Select(y => new TransactionInterDTO() { 
+                                                                Id = y.Id,
+                                                                Sum = y.Sum,
+                                                                Date = y.Date,
+                                                                Method = y.Method,
+                                                                Employee = Context.Employees.Where(z => z.PersonalCode == y.Pharmacist.PersonalCode).First().FirstName
+                                                            }).First(),
+                                        Pharmacy = Context.Pharmacy.Where(y => y.Id == x.Id).First().Address,
+                                        Warehouse = Context.Warehouse.Where(y => y.Id == x.Id).First().Address,
+                                        Provider = Context.MedicineProvider.Where(y => y.Id == x.Id).First().Name
+
+                                }).ToList()
+                }).ToListAsync();           
+
+            return Ok(new GetOrdersDTO<ProductBalanceInterDTO>(orders));
+        }
```

Comment:
This method is too convoluted and this information will not be used by frontend. Maybe we could instead make a method to retrieve a single order data? This would include Order data as well as related Product Balances data.

Reply:
Leave this for future.